lib/chef/knife/azurerm_server_create.rb

```diff
@@ -131,6 +131,10 @@ class AzurermServerCreate < Knife
                                       Do a \"knife azure image list --azure-api-mode ARM\" to see a list of available Versions.",
         :default => 'latest'
 
+      option :azure_image_os_type,
```

Comment:
Well, it's actually optional, right? If you specify the other parameters, you don't need to specify this. I think we should add an error message if you specify this parameter AND you specify offer or sku, assuming SKU is equivalent to something like "Ubuntu".